### PR TITLE
layers/meta-balena-jetson: Add new board: Auvdidea JN30D-Nano and JN30D-TX2

### PR DESCRIPTION
**Pull request by compiaffe**
Add support for Auvidea JN30D-Nano and JN30D-TX2 boards.

A break out board bundled with the NVIDIA Jetson Nano or TX2 and similar to the
dev kit.

Changelog-entry: Add support for the Auvidea JN30D-Nano and JN30D-TX2 board.

### CODE REVIEW

**Comment by acostach** [comment]
Hi @compiaffe , no need for a new coffee file and yocto machine, having the dtb in the rootfs should suffice because you can then select it in the dashboard. To test this on your board you can place the dtb in /mnt/sysroot/active/current/boot/ on a generic image from the dashboard and then specify its name in the device configuration panel. If it works fine you can then include it in your PR either for the nano sd or the emmc, or for both.

**Comment by compiaffe** [comment]
@acostach I don't quite understand. Is this a new way of supporting new break-out boards?

This is a new breakout board, very similar to the devkit, but not quite the same.

**Comment by compiaffe** [comment]
For testing purpose the instructions are clear though.

Can I specify multiple dtb files here? The JNX30D for the TX2 NX comes with multiple dtbs.

**Comment by compiaffe** [comment]
Ok, so basically this is the preferred way forward to support new TX2 and Nano breakout boards?

Can I specify the dtb file to load in a config file that I would inject?

**Comment by acostach** [comment]
@compiaffe 
> Ok, so basically this is the preferred way forward to support new TX2 and Nano breakout boards?

Yes

> Can I specify the dtb file to load in a config file that I would inject?




The dtb name is stored in resin-boot/extra_uEnv.txt, the supervisor should pick that one up as default and set it in the API while registering during the first boot. Basically if you set it in that file and then provision the device with that image, it should get populated in device configuration tab.

**Comment by compiaffe** [comment]
@acostach I can confirm the board works as far as I can tell. USB3 works fine, Ethernet as well. Deploys nicely to Balena.

**Comment by acostach** [comment]
@resin-jenkins test this please

**Comment by acostach** [comment]
@balena-ci retest

**Review by acostach — APPROVED** [review]
lgtm

**Comment by acostach** [comment]
@balena-ci retest

**Comment by acostach** [comment]
@compiaffe the commit message has a newline between Shangelog-entry and Signed-off-by, which needs to be removed for VersionBot to verify it.  While fixing the commit message here could you please include the changes from https://github.com/balena-os/balena-jetson/pull/400 here too?

**Comment by compiaffe** [comment]
@acostach Sure, all done.
Thanks for the feedback.

**Comment by acostach** [comment]
@resin-jenkins test this please

**Comment by acostach** [comment]
@balena-ci retest

**Comment by acostach** [comment]
@compiaffe there's a build error for the device type with the yocto machine named 'jetson-tx2'

```
NOTE: recipe libcap-2.51-r0: task do_populate_sysroot_setscene: Succeeded
NOTE: recipe kmod-29-r0: task do_populate_sysroot_setscene: Succeeded
WARNING: jetson-dtbs-1.0-r0 do_fetch: Failed to fetch URL [file://tegra186-p3636-0001-p3509-0000-a01-auvidea-jn30d.dtb](file:///), attempting MIRRORS if available
NOTE: Running task 446 of 5028 (/work/build/../layers/poky/meta/recipes-kernel/make-mod-scripts/make-mod-scripts_1.0.bb:do_unpack)
NOTE: recipe intltool-native-0.51.0-r0: task do_populate_sysroot_setscene: Succeeded
NOTE: recipe shadow-4.9-r0: task do_populate_sysroot_setscene: Succeeded
NOTE: recipe coreutils-native-8.32-r0: task do_populate_sysroot_setscene: Succeeded
ERROR: jetson-dtbs-1.0-r0 do_fetch: Fetcher failure: Unable to find file [file://tegra186-p3636-0001-p3509-0000-a01-auvidea-jn30d.dtb](file:///) anywhere. The paths that were searched were:
    /work/build/../layers/meta-balena-jetson/recipes-kernel/linux/linux-tegra/balena-os
```

File: layers/meta-balena-jetson/recipes-kernel/linux/linux-tegra_%.bbappend [hunk 1]
```diff
@@ -75,6 +76,10 @@ SRC_URI:append:cnx100-xavier-nx = " \
     file://tegra194-xavier-nx-cnx100.dtb \
 "
 
+SRC_URI:append:jetson-tx2-nx-devkit = " \
+    file://tegra186-p3636-0001-p3509-0000-a01-auvidea-jn30d.dtb \
+"
+
```
Comment (compiaffe) [inline layers/meta-balena-jetson/recipes-kernel/linux/linux-tegra_%.bbappend:82]: @acostach  We did not have this section at all. Can you confirm if the `jetson-tx2-nx-devkit` device is the correct device for a emmc based tx2?

**Comment by compiaffe** [comment]
@acostach that one is fixed.

**Comment by acostach** [comment]
@balena-ci retest

**Comment by acostach** [comment]
@resin-jenkins test this please

File: layers/meta-balena-jetson/recipes-kernel/linux/linux-tegra_%.bbappend (hunk 1 above)
Comment (acostach) [reply layers/meta-balena-jetson/recipes-kernel/linux/linux-tegra_%.bbappend:82]: @compiaffe Yes, the jetson-tx2-nx-devkit is for the eMMC Jetson TX2 NX module in the Xavier NX Devkit (or some other carrier board depending on the custom dtb like for this case)

**Comment by compiaffe** [comment]
@acostach Any idea when this change is going to hit the `Nvidia Jetson TX2 NX (with Xavier NX Devkit)` device on balena? They are still at `2.98.12`.

**Comment by acostach** [comment]
@compiaffe v2.105.2+rev1 is now available for the TX2 NX in balena-cloud.

**Pull request merged**